mr/src/main/java/org/apache/iceberg/mr/hive/serde/objectinspector/IcebergTimeObjectInspector.java

```diff
@@ -49,8 +50,11 @@ public Text getPrimitiveWritableObject(Object o) {
   }
 
   @Override
-  public Object convert(Object o) {
-    return o == null ? null : o.toString();
+  public LocalTime convert(Object o) {
+    if (o == null) {
+      return null;
+    }
+    return LocalTime.parse(o.toString());
```

Comment:
nit: Could you make this a one liner as all of the others?

Reply:
Fixed.